### DISCUSSION TITLE
Multiple providers

### DISCUSSION THREAD

**Discussion by gaurishhs** [go-pkgz/auth#164] [Q&A]
Hi! Is it possible that a user can log in through multiple providers? For example i signed up using Google and later i logged in with direct provider with password? How can i persist the ID of the user

**Comment by paskal** [comment]
Apologies for taking so long to answer. Coupling multiple IDs is not possible as of now. The user is uniquely identified by the provider they used to log in; new provider means new ID.

What is the use case you want to cover? If someone identifies as "User A" and keeps their reputation while commenting on some website, they can have the same name when logged in through a different provider. As we don't have mentions working in Remark42, different IDs won't be that visible to the other users, as they still see the same username.

Thinking about it from the Remark42 developer perspective, we'll open a can of worms when allowing this functionality. However, it's doable if we would want to make it happen.
